From 16cb7ff208a61217a55a656bde0060ef0b5b3de9 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:20:46 -0400
Subject: [fs] nfsd: remove nfs4_lock_state: nfs4_state_shutdown_net

Message-id: <1410898849-22126-163-git-send-email-bfields@redhat.com>
Patchwork-id: 95768
O-Subject: [Patch RHEL7 162/165] nfsd: remove nfs4_lock_state: nfs4_state_shutdown_net
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Jeff Layton <jlayton@primarydata.com>

Signed-off-by: Jeff Layton <jlayton@primarydata.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 74cf76df0f7fa39f4d9d6c43487774d43b31eb80
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index f5d434c..b900af8 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -6406,7 +6406,6 @@ nfs4_state_shutdown_net(struct net *net)
  cancel_delayed_work_sync(&nn->laundromat_work);
  locks_end_grace(&nn->nfsd4_manager);
 
- nfs4_lock_state();
  INIT_LIST_HEAD(&reaplist);
  spin_lock(&state_lock);
  list_for_each_safe(pos, next, &nn->del_recall_lru) {
@@ -6423,7 +6422,6 @@ nfs4_state_shutdown_net(struct net *net)
 
  nfsd4_client_tracking_exit(net);
  nfs4_state_destroy_net(net);
- nfs4_unlock_state();
 }
 
 void
-- 
1.7.1